Library/Homebrew/utils/github.rb

```diff
@@ -516,52 +516,75 @@ def self.pull_request_title_regex(name, version = nil)
     /(^|\s)#{Regexp.quote(name)}(:|,|\s)(.*\s)?#{Regexp.quote(version)}(:|,|\s|$)/i
   end
 
+  sig {
+    params(name: String, tap_remote_repo: String, state: T.nilable(String), version: T.nilable(String))
+      .returns(T::Array[T::Hash[String, T.untyped]])
+  }
   def self.fetch_pull_requests(name, tap_remote_repo, state: nil, version: nil)
     regex = pull_request_title_regex(name, version)
     query = "is:pr #{name} #{version}".strip
 
-    issues_for_formula(query, tap_remote_repo:, state:).select do |pr|
-      pr["html_url"].include?("/pull/") && regex.match?(pr["title"])
+    # Unauthenticated users cannot use GraphQL so use search REST API instead.
+    # Limit for this is 30/minute so is usually OK unless you're spamming bump PRs (e.g. CI).
+    if API.credentials_type == :none
+      return issues_for_formula(query, tap_remote_repo:, state:).select do |pr|
+        pr["html_url"].include?("/pull/") && regex.match?(pr["title"])
+      end
     end
-  rescue API::RateLimitExceededError => e
-    opoo e.message
-    []
-  end
 
-  # WARNING: The GitHub API returns results in a slightly different form here compared to `fetch_pull_requests`.
-  def self.fetch_open_pull_requests(name, tap_remote_repo, version: nil)
-    return [] if tap_remote_repo.blank?
-
-    # Bust the cache every three minutes.
-    cache_expiry = 3 * 60
-    cache_epoch = Time.now - (Time.now.to_i % cache_expiry)
-    cache_key = "#{tap_remote_repo}_#{cache_epoch.to_i}"
-
-    @open_pull_requests ||= {}
-    @open_pull_requests[cache_key] ||= begin
-      owner, repo = tap_remote_repo.split("/")
-      endpoint = "repos/#{owner}/#{repo}/pulls"
-      query_parameters = ["state=open", "direction=desc"]
-      pull_requests = []
+    query += " repo:#{tap_remote_repo} in:title"
+    query += " state:#{state}" if state.present?
+    graphql_query = <<~EOS
```

Comment:
Nit: This probably makes more sense as a constant.

Reply:
`frozen_string_literal` already makes this a single memory object and it is very specific to this method itself so making it global feels weird.